web/src/compose_closed_ui.ts

```diff
@@ -95,6 +98,16 @@ export function get_recipient_label(
         if (selected_message?.is_stream) {
             return get_stream_recipient_label(selected_message.stream_id, selected_message.topic);
         }
+        const recipients = selected_message.display_recipient;
+        if (
+            recipients &&
+            recipients.length === 1 &&
+            typeof recipients[0] !== "string" &&
+            recipients[0]?.id &&
+            people.is_my_user_id(recipients[0].id)
```

Comment:
For this case, we have a Message object, so is there no easier way to get this information without all of these conditionals?